clang/lib/Driver/Compilation.cpp

```diff
@@ -60,13 +60,64 @@ Compilation::~Compilation() {
       delete Arg.second;
 }
 
+static void HandleXarchArgs(DerivedArgList *OffloadArgList, const Driver &D) {
+  if (!OffloadArgList || (!OffloadArgList->hasArg(options::OPT_Xarch_device) &&
+                          !OffloadArgList->hasArg(options::OPT_Xarch_host)))
+    return;
+
+  bool IsDevice = OffloadArgList->hasArg(options::OPT_Xarch_device);
```

Comment:
Does this handle the possibility of passing both `-Xarch_host` and `-Xarch_device` on the same command line?

Reply:
Hi, @mdtoguchi 
Thanks for pointing out this, I have updated patch to handle combination use of -Xarch_device and -Xarch_host and also added lit tests.
Thanks very much.